### PR TITLE
Build for Apple Silicon in Github Actions

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -48,14 +48,7 @@ jobs:
...
         if: matrix.os == 'macOS-latest'
-        run: |
-          mkdir -p build/jni/${{ matrix.arch }}/
-          mkdir -p zipline/src/jvmMain/resources/jni/${{ matrix.arch }}/
-          cmake -S zipline/src/jvmMain/ -B build/jni/${{ matrix.arch }}/ \
-            -DQUICKJS_VERSION="$(cat zipline/native/quickjs/VERSION)" \
-            -DCMAKE_SYSTEM_PROCESSOR=${{ matrix.cmake-arch }}
-          cmake --build build/jni/${{ matrix.arch }}/ --verbose
-          cp -v build/jni/${{ matrix.arch }}/libquickjs.* zipline/src/jvmMain/resources/jni/${{ matrix.arch }}/
+        run: ./.github/workflows/build-mac.sh -a ${{ matrix.arch }} -c ${{ matrix.cmake-arch }}
```
Comment (swankjesse) [inline .github/workflows/release.yaml:51]: This is consistent with `build.yaml`.

File: .github/workflows/build-mac.sh [hunk 2]
```diff
@@ -56,7 +56,7 @@ function build() {
   echoRun mkdir -p zipline/src/jvmMain/resources/jni/$MACOS_ARCH/
   echoRun cmake -S zipline/src/jvmMain/ -B build/jni/$MACOS_ARCH/ \
     -DQUICKJS_VERSION="$(cat zipline/native/quickjs/VERSION)" \
-    -DCMAKE_SYSTEM_PROCESSOR=$CMAKE_ARCH
+    -DCMAKE_OSX_ARCHITECTURES=$CMAKE_ARCH
```
Comment (swankjesse) [inline .github/workflows/build-mac.sh:59]: I determined experimentally that this is the variable I want to change.
https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_PROCESSOR.html
https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_ARCHITECTURES.html

File: .github/workflows/release.yaml (hunk 1 above)
Comment (adrw) [reply .github/workflows/release.yaml:51]: Whoops, I missed updating the release to match build

**Review by adrw — APPROVED** [review]

**Comment by adrw** [comment]
Does this close #690?

**Comment by swankjesse** [comment]
YES.

**Pull request merged**